### PR TITLE
Router direct path

### PR DESCRIPTION
**Pull request by phudtran**
Adding router direct path feature, and its functional test.

### CODE REVIEW

File: src/xdp/trn_transit_xdp.c [hunk 1]
```diff
@@ -136,6 +137,35 @@ transit switch of that network, OW forward to the transit router. */
...
+		pkt->rts_opt->rts_data.host.ip = pkt->ip->daddr;
+		__builtin_memcpy(pkt->rts_opt->rts_data.host.mac,
+				 pkt->eth->h_dest, 6 * sizeof(unsigned char));
+	}
+
+	if (dst_r_ep) {
+		if (!pkt->ip->ttl)
+			return XDP_DROP;
+		bpf_debug(
+			"[Transit:%ld.0x%x] Host of 0x%x, found sending directly from the router!\n",
```
Comment (zasherif) [inline src/xdp/trn_transit_xdp.c]: There is no actually a router. Change to "found sending directly".

File: src/xdp/trn_transit_xdp.c [hunk 2]
```diff
@@ -136,6 +137,35 @@ transit switch of that network, OW forward to the transit router. */
...
 	net = bpf_map_lookup_elem(&networks_map, &nkey);
 
+	/* Cache lookup for known ep */
+	struct remote_endpoint_t *dst_r_ep;
+	struct endpoint_key_t dst_epkey;
+	__builtin_memcpy(&dst_epkey.tunip[0], &tunnel_id, sizeof(tunnel_id));
+	dst_epkey.tunip[2] = inner_dst_ip;
+	dst_r_ep = bpf_map_lookup_elem(&ep_host_cache, &dst_epkey);
+
+	/* Only rewrite RTS and update cache when router host */
```
Comment (zasherif) [inline src/xdp/trn_transit_xdp.c]: minor remove "when router host", there is no actual router here.

File: src/xdp/trn_transit_xdp.c (hunk 1 above)
Comment (phudtran) [reply src/xdp/trn_transit_xdp.c]: fixed.

File: src/xdp/trn_transit_xdp.c (hunk 2 above)
Comment (phudtran) [reply src/xdp/trn_transit_xdp.c]: fixed

**Pull request merged**